### PR TITLE
Add Otterdog Configuration Workflow Section

### PR DESCRIPTION
**Pull request by tsande16**
This PR introduces the new Otterdog Configuration Workflow section as well as a few minor fixes.

### CODE REVIEW

File: infrastructure-documenation/operations-production/ops-eclipse.md [hunk 1]
```diff
@@ -16,7 +16,21 @@ These [.eclipsefdn](https://github.com/eclipse-pass/.eclipsefdn) repo / tools gi
...
+To make changes to the Otterdog configuration in the [.eclipsefdn](https://github.com/eclipse-pass/.eclipsefdn)
+repository, follow these steps:
+
+1. Fork the [.eclipsefdn](https://github.com/eclipse-pass/.eclipsefdn) repository into your own GitHub account.
+2. Make changes to the `eclipse-pass.jsonnet` file.
+3. Push those changes to the upstream repository.
+4. Create a pull request in the `.eclipsefdn` repository. 
+5. An automated workflow will run, displaying the changes to be applied and validating that the configuration is 
+correctly formatted and structured.
+6. Depending on the type of changes one or more Eclipse engineers will review the PR. Additionally, the project lead
```
Comment (rpoet-jh) [inline infrastructure-documenation/operations-production/ops-eclipse.md:32]: I think `Additionally, the project lead must also sign off on it.` depends on the type of change too.

Comment (tsande16) [reply infrastructure-documenation/operations-production/ops-eclipse.md:32]: This comment from an old PR of mine makes me think they need to sign off on it: https://github.com/eclipse-pass/.eclipsefdn/pull/5#issuecomment-1969123118

I can't find anything in the Eclipse handbook. I will take a look at other Eclipse issues and see if I can find anything. If I can't find anything I will open up a ticket to get clarification on this.

Comment (tsande16) [reply infrastructure-documenation/operations-production/ops-eclipse.md]: Sounds good. I will open up an Eclipse help ticket asking about this.

Additionally, after reading some of the Eclipse Handbook I found some interesting notes about what should be in our `contributing.md` file

Comment (tsande16) [reply infrastructure-documenation/operations-production/ops-eclipse.md]: Okay, before submitting my help ticket I came across this information:

`Normally we ask for additional approval from a project-lead, but it is up to each project to decide how they would like to handle it. That is just informal for now, but we would like to make this configurable in the repo itself for future PRs.`
[Source](https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues/3514#note_1189953)

So it sounds like it's up to us how we want these approved. Do you want me to add this to our next PASS status meeting discussion, or do you think it's okay how it's worded as now?

Comment (rpoet-jh) [reply infrastructure-documenation/operations-production/ops-eclipse.md]: Interesting.  I would say to make it simple, Project Lead approval should always be required for .eclipsefdn PRs, but we can talk about it in the next PASS status meeting.  The wording is fine as is for now.

**Pull request merged**